clusterloader2/pkg/prometheus/prometheus.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package prometheus
+
+import (
+	"fmt"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/config"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework/client"
+	"os"
+	"path/filepath"
+)
+
+const (
+	namespace = "monitoring"
+)
+
+// SetUpPrometheusStack sets up prometheus stack in the cluster.
+// It does nothing if the prometheus stack is already set up in the cluster.
+func SetUpPrometheusStack(framework *framework.Framework) error {
+	k8sClient := framework.GetClientSets().GetClient()
+	if isPrometheusStackSetUp, err := isPrometheusStackSetUp(k8sClient); err != nil {
+		return err
+	} else if isPrometheusStackSetUp {
+		klog.Info("Prometheus stack already set up.")
+		return nil
+	}
+
+	klog.Info("Setting up prometheus stack")
+	if err := client.CreateNamespace(k8sClient, namespace); err != nil {
+		return err
+	}
+	klog.Infof("Namespace %s created", namespace)
+	if err := applyManifests(framework); err != nil {
+		return err
+	}
+	klog.Info("Prometheus stack set up successfully")
+	return nil
+}
+
+func isPrometheusStackSetUp(client clientset.Interface) (bool, error) {
+	_, err := client.CoreV1().Namespaces().Get(namespace, metav1.GetOptions{})
```

Comment:
Retry missing. Example: https://github.com/kubernetes/perf-tests/blob/6cde98fdcc96aaff7e0c12f6c4b6bfb767362497/clusterloader2/pkg/framework/client/objects.go#L100
This namespace getter can be added to "k8s.io/perf-tests/clusterloader2/pkg/framework/client".

Reply:
Done.